Tests/FluentAssertions.Specs/Execution/TestFrameworkProviderTests.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using FluentAssertions.Common;
+using FluentAssertions.Execution;
+using Xunit;
+
+using static FluentAssertions.FluentActions;
+
+namespace FluentAssertions.Specs.Execution;
+
+public class TestFrameworkProviderTests
+{
+    [Fact]
+    public void When_running_xunit_test_implicitly_it_should_be_detected()
+    {
+        var configuration = new Configuration(new TestConfigurationStore());
+        var result = TestFrameworkProvider.DetectFramework(configuration);
+
+        result.IsAvailable.Should().BeTrue();
+        result.Should().BeOfType<XUnit2TestFramework>();
+    }
+
+    [Fact]
+    public void When_running_xunit_test_explicitly_it_should_be_detected()
+    {
+        var configuration = new Configuration(new TestConfigurationStore())
+        {
+            TestFrameworkName = "xunit2"
+        };
+        var result = TestFrameworkProvider.DetectFramework(configuration);
+
+        result.IsAvailable.Should().BeTrue();
+        result.Should().BeOfType<XUnit2TestFramework>();
+    }
+
+    [Fact]
+    public void When_running_test_with_unknown_test_framework_it_should_throw()
+    {
+        var configuration = new Configuration(new TestConfigurationStore())
+        {
+            TestFrameworkName = "foo"
+        };
+
+        Invoking(() => TestFrameworkProvider.AttemptToDetectUsingAppSetting(configuration))
+            .Should().Throw<InvalidOperationException>()
+            .WithMessage("FluentAssertions was configured to use the test framework 'foo' but this is not supported."
+                         + " Please use one of the supported frameworks: mspec, nspec3, nunit, mstestv2, xunit2.");
```

Comment:
Just a suggestion to rely a little less on the _exact_ failure message.

```cs
.WithMessage("*the test framework 'foo' but this is not supported*");
```

Reply:
done